pyfarm/agent/service.py

```diff
@@ -443,6 +444,14 @@ def wait_on_stopped():
             wait_on_stopped()
 
     def sigint_handler(self, *_):
+        try:
+            os.remove(config["run_control_file"])
+        except (WindowsError, OSError, IOError) as e:
+            if e.errno != ENOENT:
+                logger.error("Could not delete run control file %s: %s: %s",
+                             config["run_control_file"],
+                             type(e).__name__, e)
+
```

Comment:
As said above, I don't see why it would work any better as an exit handler than right here.